_api-reference/snapshots/get-snapshot-status.md

```diff
@@ -23,24 +23,32 @@ Path parameters are optional.
 | Parameter | Data type | Description | 
 :--- | :--- | :---
 | repository | String | Repository containing the snapshot. |
-| snapshot | String | Snapshot to return. |
+| snapshot | List | Snapshot(s) to return. |
+| index | List | Indexes to include in the response. |
 
 Three request variants provide flexibility:
 
 * `GET _snapshot/_status` returns the status of all currently running snapshots in all repositories.
 
 * `GET _snapshot/<repository>/_status` returns all currently running snapshots in the specified repository. This is the preferred variant.
 
-* `GET _snapshot/<repository>/<snapshot>/_status` returns detailed status information for a specific snapshot in the specified repository, regardless of whether it's currently running or not. 
+* `GET _snapshot/<repository>/<snapshot>/_status` returns detailed status information for a specific snapshot(s) in the specified repository, regardless of whether it's currently running or not. 
 
-Using the API to return state for other than currently running snapshots can be very costly for (1) machine machine resources and (2) processing time if running in the cloud. For each snapshot, each request causes file reads from all a snapshot's shards. 
+* `GET /_snapshot/<repository>/<snapshot>/<index>/_status` returns detailed status information only for the specified indexes in a specific snapshot for the specified repository. Note that this endpoint works only for indexes of a specific snapshot.
+
+Snapshot API calls only work if the total number of shards across the requested resources, such as snapshots and indexes created from snapshots, is less than the limit specified by the following cluster setting:
+
+- `snapshot.max_shards_allowed_in_status_api`(Dynamic, integer): The maximum number of shards that can be included in the snapshot status API response. Default value is `200000`. Not applicable for [Shallow V2 Snapshots]({{site.url}}{{site.baseurl}}/tuning-your-cluster/availability-and-recovery/remote-store/snapshot-interoperability##shallow-snapshot-v2), where the total number and size of files are returned as 0. 
+
+
+Using the API to return state for other than currently running snapshots can be very costly for (1) machine resources and (2) processing time if running in the cloud. For each snapshot, each request causes file reads from all of the snapshot's shards. 
```

Comment:
I'm not following these sentences. Do we mean "Using the API to return the state of snapshots that are not currently running can be very costly in terms of 1) machine resources and 2) processing time if you are running your [noun] in the cloud. For each snapshot, each request causes a file read from all of the snapshot's shards"?

Reply:
This one stumped me as well. In my testing, the Snapshot Status API returns information on all shards. This rewrite sounds better to me but does it get the point across for you @ltaragi?

Reply:
@Naarcha-AWS Please use the edited text in my comment above. Thanks!